File: code/unclassified/src/palindrome/palindrome_check/palindrome_check.js
/*

// unclassified | palindrome | palindrome check | palindrome check | JAVASCRIPT
// Part of Cosmos by OpenGenus Foundation

*/


function isPalindrome(str) {
  return [...str].reverse().join("") == str;
}

console.log(isPalindrome("lol"));
console.log(isPalindrome("nitin"));
console.log(isPalindrome("terabyte"));
console.log(isPalindrome("tbhaxor"));

// Output
// true
// true
// false
// false

File: code/unclassified/src/palindrome/palindrome_check/palindrome_check.cs
using System;

namespace Palindrome
{
    class Program
    {
        public static bool isPalindrome(string str)
        {
            for (int i = 0, j = str.Length - 1; i < str.Length / 2; ++i, --j)
            {
                if(str[i] != str[j])
                {
                    return false;
                }
            }
            return true;
        }

        public static void Main()
        {
            Console.WriteLine(isPalindrome("lol")); // True
            Console.WriteLine(isPalindrome("lolwa"));  // False
        }
    }
}